controlplane/kubeadm/controllers/controller_test.go

```diff
@@ -1685,8 +1692,11 @@ func TestKubeadmControlPlaneReconciler_scaleDownControlPlane_NoError(t *testing.
 		Machines: machines,
 	}
 
-	_, err := r.scaleDownControlPlane(context.Background(), cluster, kcp, machines, machines, controlPlane)
-	g.Expect(err).ToNot(HaveOccurred())
+	ml := clusterv1.MachineList{}
+	ml.Items = []clusterv1.Machine{*machines["two"]}
+	mll := internal.NewFilterableMachineCollectionFromMachineList(&ml)
+	_, err := r.scaleDownControlPlane(context.Background(), cluster, kcp, machines, mll, controlPlane)
+	g.Expect(err).To(HaveOccurred())
```

Comment:
This expectation is now inconsistent with the test title which says `NoError`, perhaps we should keep the existing test as a NoError and make this a new test that expects an error, WDYT?